### PR TITLE
First solution

### PR DESCRIPTION
**Pull request by KIR4S3K**

### CODE REVIEW

File: src/test/java/core/basesyntax/ShopServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Test
+    void testAddFruit() {
+        shopService.addFruit("apple", 10);
+        assertEquals(10, shopService.getFruitQuantity("apple"));
+    }
+}
+
+class ShopServiceImpl {
```
Comment (slade13) [inline src/test/java/core/basesyntax/ShopServiceImplTest.java:27]: The `ShopServiceImpl` class should not be defined within the test class. It should be in its own file in the main source directory. This separation is important for maintaining a clean project structure and ensuring that the implementation code is not mixed with test code.

File: src/test/java/core/basesyntax/ShopServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    private ShopServiceImpl shopService;
+
+    @BeforeEach
+    void setUp() {
+        shopService = new ShopServiceImpl();
+    }
+
+    @Test
+    void testAddFruit() {
```
Comment (slade13) [inline src/test/java/core/basesyntax/ShopServiceImplTest.java:21]: The test method name `testAddFruit` does not follow the recommended naming convention for test methods. It should be more descriptive, such as `shouldAddFruitCorrectly` to clearly indicate what the test is verifying.

File: src/test/java/core/basesyntax/ShopServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        shopService.addFruit("apple", 10);
+        assertEquals(10, shopService.getFruitQuantity("apple"));
```
Comment (slade13) [inline src/test/java/core/basesyntax/ShopServiceImplTest.java:23]: The test only checks a single scenario for adding a fruit. Consider adding more test cases to cover different scenarios, such as adding a fruit that already exists to ensure the quantity is updated correctly.

File: src/main/java/core/basesyntax/.idea/.gitignore
Comment (slade13) [inline src/main/java/core/basesyntax/.idea/.gitignore:1]: Don't commit files from your .idea directory, you can add them to .gitignore.

File: src/main/java/core/basesyntax/.idea/misc.xml
Comment (slade13) [inline src/main/java/core/basesyntax/.idea/misc.xml:1]: Don't commit files from your .idea directory, you can add them to .gitignore.

File: src/main/java/core/basesyntax/.idea/modules.xml
Comment (slade13) [inline src/main/java/core/basesyntax/.idea/modules.xml:1]: Don't commit files from your .idea directory, you can add them to .gitignore.

File: src/main/java/core/basesyntax/.idea/vcs.xml
Comment (slade13) [inline src/main/java/core/basesyntax/.idea/vcs.xml:1]: Don't commit files from your .idea directory, you can add them to .gitignore.

**Review by slade13 — CHANGES_REQUESTED** [review]
Your code does not compile successfully, please fix it and review my comments.

**Review by slade13 — CHANGES_REQUESTED** [review]
Your code still does not compile, have you checked it?
You have to resolve following problems:
```
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/controller/FruitShopController.java:18:1: Extra separation in import group before 'java.util.ArrayList' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/strategy/impl/ReturnOperation.java:6:1: Extra separation in import group before 'java.util.Map' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java:6:1: Extra separation in import group before 'java.util.Map' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/strategy/impl/BalanceOperation.java:6:1: Extra separation in import group before 'java.util.Map' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java:6:1: Extra separation in import group before 'java.util.Map' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/strategy/OperationHandler.java:5:1: Extra separation in import group before 'java.util.Map' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/service/ShopService.java:5:1: Extra separation in import group before 'java.util.List' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/service/impl/FileReaderImpl.java:5:1: Extra separation in import group before 'java.nio.file.Files' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/service/impl/FileWriterImpl.java:5:1: Extra separation in import group before 'java.nio.file.Files' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java:7:1: Extra separation in import group before 'java.util.Collections' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java:5:1: Extra separation in import group before 'java.util.ArrayList' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/test/java/core/basesyntax/ShopServiceImplTest.java:7:1: Extra separation in import group before 'java.util.HashMap' [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/test/java/core/basesyntax/ShopServiceImplTest.java:7:1: Wrong lexicographical order for 'java.util.HashMap' import. Should be before 'org.junit.jupiter.api.Test'. [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/test/java/core/basesyntax/ShopServiceImplTest.java:8:1: Wrong lexicographical order for 'java.util.Map' import. Should be before 'org.junit.jupiter.api.Test'. [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/test/java/core/basesyntax/ShopServiceImplTest.java:8:8: Unused import - java.util.Map. [UnusedImports]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/test/java/core/basesyntax/ShopServiceImplTest.java:10:1: Import statement for 'org.junit.jupiter.api.Assertions.assertEquals' is in the wrong order. Should be in the 'STATIC' group, expecting not assigned imports on this line. [CustomImportOrder]
Error:  /home/runner/work/jv-fruit-shop-tests/jv-fruit-shop-tests/src/test/java/core/basesyntax/ShopServiceImplTest.java:11:1: Import statement for 'org.junit.jupiter.api.Assertions.assertThrows' is in the wrong order. Should be in the 'STATIC' group, expecting not assigned imports on this line. [CustomImportOrder]
````

Here is the latest build log: https://github.com/mate-academy/jv-fruit-shop-tests/actions/runs/13794817551/job/38583732212?pr=1130

**Review by slade13 — CHANGES_REQUESTED** [review]
Now your code does not build because of not enough lines covered by tests: https://github.com/mate-academy/jv-fruit-shop-tests/pull/1130/checks

It requires tests of line coverage above 80%
`Warning:  Rule violated for bundle jv-fruit-shop-tests: lines covered ratio is 0.0, but expected minimum is 0.8`

File: src/test/java/core/basesyntax/FileWriterImplTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class FileWriterImplTest {
+    private static final String TEST_FILE = "src/test/resources/testOutput.csv";
+
+    @Test
+    void write_validData_shouldCreateFile() throws Exception {
```
Comment (slade13) [inline src/test/java/core/basesyntax/FileWriterImplTest.java:15]: You could add also edge cases for writing empty list of data and file path that already exists (should it overwrite? Append?)

File: src/test/java/core/basesyntax/FileWriterImplTest.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.service.impl.FileWriterImpl;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class FileWriterImplTest {
+    private static final String TEST_FILE = "src/test/resources/testOutput.csv";
```
Comment (slade13) [inline src/test/java/core/basesyntax/FileWriterImplTest.java:12]: Consider using `@TempDir` to avoid writing to a static path (src/test/resources/...), which could interfere with source files or require cleanup.

File: src/test/java/core/basesyntax/FruitShopControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.controller.FruitShopController;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.jupiter.api.Test;
+
+class FruitShopControllerTest {
+    private static final String REPORT_FILE = "src/main/resources/finalReport.csv";
```
Comment (slade13) [inline src/test/java/core/basesyntax/FruitShopControllerTest.java:11]: Use temporary paths instead of a fixed one in src/main/resources/.

File: src/test/java/core/basesyntax/FruitShopControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+    @Test
+    void run_shouldGenerateReportFile() {
```
Comment (slade13) [inline src/test/java/core/basesyntax/FruitShopControllerTest.java:14]: Add also test for report content, not just existence.

File: src/test/java/core/basesyntax/FruitShopControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.controller.FruitShopController;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.jupiter.api.Test;
+
+class FruitShopControllerTest {
```
Comment (slade13) [inline src/test/java/core/basesyntax/FruitShopControllerTest.java:10]: You can also add test to check what happens if input data is malformed or missing.

File: src/test/java/core/basesyntax/ReportGeneratorImplTest.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+
+import core.basesyntax.service.impl.ReportGeneratorImpl;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+
+class ReportGeneratorImplTest {
```
Comment (slade13) [inline src/test/java/core/basesyntax/ReportGeneratorImplTest.java:11]: Add a test for empty map (should return just a header or empty list).

File: src/test/java/core/basesyntax/ShopServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+import core.basesyntax.strategy.impl.PurchaseOperation;
+import core.basesyntax.strategy.impl.ReturnOperation;
+import core.basesyntax.strategy.impl.SupplyOperation;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ShopServiceImplTest {
```
Comment (slade13) [inline src/test/java/core/basesyntax/ShopServiceImplTest.java:19]: Add negative test cases:
- Missing handler for an operation.
- Invalid fruit names or negative quantities.

Comment (slade13) [inline src/test/java/core/basesyntax/ShopServiceImplTest.java:19]: You could think about splitting your tests into parameterized ones to keep your code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

**Review by slade13 — APPROVED** [review]
Good job! Please review my comments. 

Always check if your code builds successfully before passing it further for code review.

Please use more descriptive commit messages and less commits if it's possible. Here you can find more information about naming your commits: https://www.baeldung.com/ops/git-commit-messages

Here you can read about interactive rebase which helps you to keep your commits clean: https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase
